### PR TITLE
added uniqueness test to resulting paths

### PR DESCRIPTION
**Pull request by kadern0**
**Fixes issue:**
#139 

**Description:**
Adds a check to ensure the resulting artifact name is unique

**Please verify and check that the pull request fulfills the following
requirements:**

- [X] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by shibumi** [comment]
LGTM.

I just realized that all of our error messages are upper-case. This is a violation of a go standard about error messages. @adityasaky we might should track this in a separate issue.

We can merge this one here

**Comment by adityasaky** [comment]
We also likely want to make it consistent with the use of `errors` across the board? We can track it in a separate issue, but IMO we can also start by using it in new code or code that's already modifying some error string?

File: in_toto/runlib.go
Comment (adityasaky) [inline in_toto/runlib.go:188]:
```suggestion
					return fmt.Errorf("left stripping has resulted in non unique dictionary key: %s", path)
```

**Review by adityasaky — APPROVED** [review]
Minor nit aside, this LGTM @kadern0, thanks! I'd like to hear from @shibumi about switching to `errors` here, but happy to merge if we want to hold off on that.

File: in_toto/runlib.go
Comment (shibumi) [reply in_toto/runlib.go:188]: I don't know. I think I would prefer to handle the error strings isolated in a new PR. We shouldn't mix code reformatting with bugfixes or features. Commits need to be atomic.

Comment (kadern0) [reply in_toto/runlib.go:188]: I can update the error string and squash both commits if you want.

Comment (shibumi) [reply in_toto/runlib.go:188]: @kadern0 no, not necessary. I really think we should address the wrong error string formatting in a different issue/PR.
Do you agree @adityasaky ?

Comment (adityasaky) [reply in_toto/runlib.go:188]: Error string formatting definitely belongs in its own issue.

Comment (adityasaky) [reply in_toto/runlib.go:188]: To be clear, @kadern0 I think the suggestion I made (or any other version of including "left strip" in the error message for clarity) is sufficient to merge this PR. Apologies for the non PR specific discussions that may have caused some confusion. I'd only like to update the error message to ensure that anyone using the functionality know _exactly_ what the issue is.

Comment (shibumi) [reply in_toto/runlib.go:188]: Ah okay. Thanks for the clarification. Can we use "Left stripping ..." in this PR then + add a new issue for solving the issues with error formatting in general?

Comment (adityasaky) [reply in_toto/runlib.go:188]: I don't think we need to worry about capitalizing "left" only to handle it again in an error strings formatting PR. Since we've (informally) decided to make that change, we can choose to apply it to any changes we make to error strings / new error strings even if the formatting hasn't been applied to the rest. I may have misunderstood you though. :smile:

I also do wonder if we can just handle everything (formatting, using `errors` consistently) under the umbrella of #5.

Comment (shibumi) [reply in_toto/runlib.go:188]: Yep, makes sense to handle this under the umbrella of #5.

> I don't think we need to worry about capitalizing "left" only to handle it again in an error strings formatting PR.

I am a fan of atomic PRs. A PR should consist only of the change necessary for solving the specified problem. Adding reformatting to it adds more "complexitiy". If we merge this PR with one error string starting with lower case, this would be inconsistent with the rest of the code base. It's better to handle this separately, imho. But feel free to merge it.

Comment (adityasaky) [reply in_toto/runlib.go:188]: I'm okay either way :smile:

Comment (shibumi) [reply in_toto/runlib.go:188]: In that case, just let us merge this and do the error string reformat under the umbrella of #5 like you suggested.

**Comment by shibumi** [comment]
Thanks @kadern0 for your work.

**Pull request merged**
